### DISCUSSION TITLE
[BUG] Please fix the wrong usage of module-alias inside zeus cli

### DISCUSSION THREAD

**Discussion by fhfuih** [zepp-health/zeppos-forum#111] [Developer Feedback]
I noticed that zeus cli uses a package called  `module-alias`, and the usage is problematic. This causes the zeus cli to abort when installed by `pnpm`

## Bug

1. Download `pnpm` in node.js environment
2. Download zeus cli using `pnpm add -g @zeppos/zeus-cli`
3. Use any zeus command
4. The following error will be thrown:

```
/Users/zeyu/Library/pnpm/global/5/.pnpm/module-alias@2.2.2/node_modules/module-alias/index.js:184
    throw new Error('Unable to find package.json in any of:\n[' + pathString + ']')
    ^

Error: Unable to find package.json in any of:
[/Users/zeyu/Library/pnpm/global/5/.pnpm/module-alias@2.2.2,
/Users/zeyu/Workspace/zeppos-samples/application/1.0/calories]
    at init (/Users/zeyu/Library/pnpm/global/5/.pnpm/module-alias@2.2.2/node_modules/module-alias/index.js:184:11)
    at Object.<anonymous> (/Users/zeyu/Library/pnpm/global/5/.pnpm/module-alias@2.2.2/node_modules/module-alias/register.js:1:13)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/Users/zeyu/Library/pnpm/global/5/.pnpm/@zeppos+zeus-cli@1.2.2/node_modules/@zeppos/zeus-cli/bin/main.js:3:1)
```

## Why it's `zeus-cil`'s fault

As stated on `module-alias`'s [official readme](https://github.com/ilearnio/module-alias#using-within-another-npm-package) regarding using this package as a dependency within another npm package, package developers should always specify the `package.json` directory instead of using `register()` to auto-detect. 

In pnpm (and presumably more future package managers), global node_modules are organized differently on the file system. This breaks some assumptions of the global package's file structures for npm users. 

## Temporary fix

In `bin/main.js`, change the following line (3)
```js
require('module-alias').register();
```

to
```js
const path = require('path')
require('module-alias')(path.resolve(__dirname, '..'));
```

Looking forward to an official update!

**Comment by MyoungXUE** [comment]
Thank you for your feedback, the relevant colleagues are already in the process of fixing.

**Comment by fhfuih** [comment]
Thanks. Just for your information, when installing Zeus-cli locally inside the project (-D) and using `pnpm exec zeus…` (essentially finding and running executables from local node_modules/bin), the error message is different. It complains to have failed to require a zeppos-..-utils file. But the fix is the same.

在 2023年3月14日，上午10:27，Myoung XUE - Zepp Health ***@***.***> 写道：

﻿

Thank you for your feedback, the relevant colleagues are already in the process of fixing.
